h/static/scripts/controllers/confirm-submit-controller.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+var Controller = require('../base/controller');
+
+/* Turn a normal submit element into one that shows a confirm dialog.
+ *
+ * The element's form will only be submitted if the user answers the
+ * confirmation dialog positively.
+ *
+ */
+class ConfirmSubmitController extends Controller {
```

Comment:
This is absolutely fine, the pattern isn't really applicable/useful in this context. It _is_ useful when the component has a non-trivial amount of state to manage, in which case it provides a pattern for how to store and update that state and how to keep the DOM in sync with that state.